java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightConstants.java

```diff
@@ -23,5 +23,6 @@
 public interface FlightConstants {
 
   String SERVICE = "arrow.flight.protocol.FlightService";
+  String PROPERTY_HEADER = "Arrow-Properties";
```

Comment:
Let's rename this to something more specific (HEADERS_KEY?) and make it an instance of `FlightServerMiddleware.Key`. Let's also change the constant to something that's more clearly namespaced under arrow, e.g. "org.apache.arrow.flight.ServerHeaderHandler".